doc/dev/tests.md

```diff
@@ -0,0 +1,378 @@
+# Setup for Testing Tutorial, Python
+In this section, we will provide the introduction to the testing framework by:
+
+- [Setting up your development environment](#setup-the-development-environment)
+- [Integrating with pytest](#integrate-with-the-pytest-test-framework)
+- [Using Tox](#tox)
+- [The `devtools_testutils` package](#devtools_testutils-package)
+- [Writing New Tests](#Writing-new-tests)
+- [Define our credentials and settings](#define-credentials)
+- [Create live test resources](#create-live-test-resources)
+- [Write our test](#writing-your-test)
+- [An example test](#an-example-test)
+- [Run and record our tests](#run-and-record-the-test)
+
+## Setup your development environment
+
+The Azure SDK Python team creates libraries that are compatible with Python 2.7 and 3.5 and up. We will set up working Python virtual environments for Python 2.7, 3.5, and 3.9. Although it is not required to test every PR against all three changes, it is useful to have a virtual environment for each version to make debugging PRs easier locally.
+
+* Python 3.9: Use the [python website](https://www.python.org/downloads/) or the one-click experience from the [Windows store](https://www.microsoft.com/en-us/p/python-38/9mssztt1n39l) (Windows only).
+* Python 3.5: Use the [python website](https://www.python.org/downloads/release/python-3510/)
+* Python 2.7: Use the [python website](https://www.python.org/downloads/release/python-2710/)
+```cmd
+C:\Users> python -m venv env
+C:\Users> env\scripts\activate       # PowerShell only
+C:\Users> source env\bin\activate    # Linux shell (Bash, ZSH, etc.) only
+C:\Users> env\scripts\activate.bat   # Windows CMD only
+(env)C:\Users>
+```
+To create virtual environment for different versions of Python use the `-p` flag to pass the specific Python executable you want to use
+```cmd
+C:\Users> python -m venv -p <path/to/Python/Python35/python.exe> py35_venv
+C:\Users> python -m venv -p <path/to/Python/Python27/python.exe> py27_venv
+```
+
+### SDK root directory
+
+In the root directory of our SDK, a number of mandatory files have been added. You will not need to edit these files for this tutorial, but you should be aware about the various components and where to find more details on them. When creating your own SDK, these files can be copied from the [`sdk/template`](https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/template) project, and modified to your needs.
+
+- README.md. This is the description and guidance for customers or your SDK. Please see the guide on writing a README to make sure you have the complete [content requirements and formatting](https://review.docs.microsoft.com/en-us/help/contribute-ref/contribute-ref-how-to-document-sdk?branch=master#readme).
```

Comment:
```suggestion
- README.md. This is the description and guidance for customers or your SDK. Please see the guide on writing a README to make sure you have the complete [content requirements and formatting](https://review.docs.microsoft.com/help/contribute-ref/contribute-ref-how-to-document-sdk?branch=master#readme).
```

Reply:
good catch, removing a few other en-us in links